src/commands/get-woocommerce-stats.php

```diff
@@ -0,0 +1,219 @@
+				Example usage:\n
+				get-woocommerce-stats --unit=year --date=2022\n
+				get-woocommerce-stats --unit=week --date=2022-W12\n
+				get-woocommerce-stats --unit=month --date=2021-10\n
+				get-woocommerce-stats --unit=day --date=2022-02-27"
+			)
+			->addOption(
+				'unit',
+				null,
+				InputOption::VALUE_REQUIRED,
+				'Options: day, week, month, year.'
+			)
+			->addOption(
+				'date',
+				null,
+				InputOption::VALUE_REQUIRED,
+				'Options: YYYY-MM-DD, YYYY-W##, YYYY-MM, YYYY.'
+			)
+			->addOption(
+				'check-production-sites',
+				null,
+				InputOption::VALUE_NONE,
+				'Checks production sites instead of the Jetpack Profile for the sites. Takes much longer to run.'
+			);
+	}
+
+	protected function execute( InputInterface $input, OutputInterface $output ) {
+		$plugin_slug = 'woocommerce';
+
+		$api_helper = new API_Helper();
+
+		// error if the unir or date options are not set
+		if ( empty( $input->getOption( 'unit' ) ) ) {
+			$output->writeln( '<error>Time unit is required for fetching stats. (example: --unit=year)</error>' );
+			exit;
+		}
+
+		if ( empty( $input->getOption( 'date' ) ) ) {
+			$output->writeln( '<error>Date is required for fetching stats (example: --date=2021-08)</error>' );
+			exit;
+		}
+
+		$unit = $input->getOption( 'unit' );
+		$date = $input->getOption( 'date' );
+
+		$output->writeln( '<info>Fetching production sites connected to a8cteam51...<info>' );
+
+		// Fetching sites connected to a8cteam51
+		$sites = $api_helper->call_wpcom_api( 'rest/v1.1/jetpack-blogs/', array() );
+
+		if ( empty( $sites ) ) {
+			$output->writeln( '<error>Failed to fetch sites.<error>' );
+			exit;
+		}
+
+		// Filter out non-production sites
+		$site_list = array();
+		foreach ( $sites->blogs->blogs as $site ) {
+			if ( strpos( $site->siteurl, 'mystagingwebsite.com' ) === false && strpos( $site->siteurl, 'go-vip.co' ) === false && strpos( $site->siteurl, 'wpcomstaging.com' ) === false && strpos( $site->siteurl, 'wpengine.com' ) === false && strpos( $site->siteurl, 'jurassic.ninja' ) === false ) {
+				$site_list[] = array( $site->userblog_id, $site->siteurl );
+			}
+		}
+		$site_count = count( $site_list );
+
+		$output->writeln( "<info>{$site_count} sites found.<info>" );
+		$output->writeln( '<info>Checking each site for WooCommerce...<info>' );
+
+		if ( $input->getOption( 'check-production-sites' ) ) {
+			$output->writeln( '<info>Checking production sites for WooCommerce...<info>' );
+			$progress_bar = new ProgressBar( $output, $site_count );
+			$progress_bar->start();
+			// Checking each site for the plugin slug: woocommerce, and only saving the sites that have it active
+			foreach ( $site_list as $site ) {
+				$progress_bar->advance();
+				$plugin_list = $this->get_list_of_plugins( $site[0] );
+				if ( ! is_null( $plugin_list ) ) {
+					if ( ! is_null( $plugin_list->data ) ) {
+						$plugins_array = json_decode( json_encode( $plugin_list->data ), true );
```

Comment:
What's the reasoning for encoding then immediately decoding? Is the intent to convert data from an object to an array or something else?